deployer/commands/transform/cost_table.py

```diff
@@ -0,0 +1,163 @@
+import re
+from pathlib import Path
+
+import pandas as pd
+import typer
+
+from deployer.cli_app import transform_app
+from deployer.utils.rendering import print_colour
+
+# Creates a new typer application, which is then nested as a sub-command named
+# "cost-table" under the `transform` sub-command of the deployer.
+cost_table_app = typer.Typer(pretty_exceptions_show_locals=False)
+transform_app.add_typer(
+    cost_table_app,
+    name="cost-table",
+    help="Transform the cost table from a cloud provider when completing billing cycles",
+)
+
+
+@cost_table_app.command()
+def aws(
+    input_path: Path = typer.Argument(
+        ..., help="The file path to the cost table downloaded from the AWS UI"
+    ),
+    output_path: Path = typer.Option(
+        None,
+        help="(Optional) The path to write the output CSV to. If None, one will be constructed.",
+    ),
+):
+    """
+    Ingests a CSV cost table generated via the AWS UI and performs a transformation.
+
+    We assume the input CSV has a column for each linked account and rows for:
+    - the linked account ID and name
+    - a monthly total for each month selected
+    - a linked account total across the month range
+
+    We aim to have an output CSV file with columns:
+    - linked account name
+    - start-month
+    - ...
+    - end-month
+    - linked account total
+    """
+    # Read the CSV file into a pandas dataframe. Skip the first row and this
+    # contains numerical project IDs - the project names begin on the second row.
+    # Rename the columns so that they are all lower case and any whitespace is
+    # replaced with underscores. Then strip '_($)' to ensure we just have the
+    # project names.
+    df = pd.read_csv(
+        input_path,
+        skiprows=1,
+    ).rename(columns=lambda col: col.strip().lower().replace(" ", "_").strip("_($)"))
+
+    # Ensure values of the linked_account_name column are lower case and any
+    # whitespace is replaced with underscores.
+    df["linked_account_name"] = df["linked_account_name"].apply(
+        lambda val: val.strip().lower().replace(" ", "_")
```

Comment:
spaces are allowed here apparently (https://docs.aws.amazon.com/organizations/latest/APIReference/API_Account.html). But am curious about the reason for us doing this transformation. Wouldn't it cause confusion if we were ever to look for the account name in the UI based on what's here?

Reply:
This is a quirk of how AWS outputs the data, how pandas infers columns, and the fact we haven't transposed the dataframe yet.

AWS outputs the linked account name as a _row_, but pandas reads the first row as column headers. So this line of code is actually affecting the column that has the months in it and some other random stuff we don't care about, not the account names.

<img width="383" alt="Screenshot 2024-05-14 at 10 26 21" src="https://github.com/2i2c-org/infrastructure/assets/44771837/09ec5b6d-131b-4cf6-9d6e-e37cb7562192">

Reply:
I expanded the comment [`3458571` (#3992)](https://github.com/2i2c-org/infrastructure/pull/3992/commits/345857182d86e5879061c6b4f44b51aa0ec116f9)

Reply:
I also pushed [`96f4dfd` (#3992)](https://github.com/2i2c-org/infrastructure/pull/3992/commits/96f4dfdb04eefaad3c1051dfd3031fb4834604ab) to address your concern that in the future we may have an account name with whitespace in it

Reply:
Generally with pandas, it's nice not to have whitespace in column names and indices, because then syntax like `df.column_name` works, as well as `df["column_name"]` - but this _only_ works if there is no whitespace.